components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -291,22 +294,65 @@ public void updateAPI(API.APIBuilder apiBuilder) throws APIManagementException {
-                    apiBuilder.lifecycleState(getApiLifecycleManager().checkListItemEvent(api.getLifecycleInstanceId
-                            (), status, checkListItem.getKey(), checkListItem.getValue()));
+                    if (APIMgtConstants.DEPRECATE_PREVIOUS_VERSIONS.equals(checkListItem.getKey())) {
+                        deprecateOlderVersion = checkListItem.getValue();
+                    } else if (APIMgtConstants.REQUIRE_RE_SUBSCRIPTIONS.equals(checkListItem.getKey())) {
+                        requireReSubscriptions = checkListItem.getValue();
+                    } else {
+                        apiBuilder.lifecycleState(getApiLifecycleManager().checkListItemEvent(api.getLifecycleInstanceId
+                                (), status, checkListItem.getKey(), checkListItem.getValue()));
+                    }
                 }
                 API originalAPI = apiBuilder.build();
                 getApiLifecycleManager().executeLifecycleEvent(status, apiBuilder
                         .getLifecycleInstanceId(), getUsername(), originalAPI);
+                if (deprecateOlderVersion) {
+                    if (StringUtils.isNotEmpty(api.getPreviousId())) {
+                        API oldAPI = getApiDAO().getAPI(api.getPreviousId());
+                        if (oldAPI != null) {
+                            API.APIBuilder previousAPI = new API.APIBuilder(oldAPI);
+                            previousAPI.setLifecycleStateInfo(getApiLifecycleManager().getCurrentLifecycleState
+                                    (previousAPI.getLifecycleInstanceId()));
+                            getApiLifecycleManager().executeLifecycleEvent(APIStatus.DEPRECATED.getStatus(), previousAPI
+                                    .getLifecycleInstanceId(), getUsername(), previousAPI.build());
+                        }
+                    }
+                }
+                if (requireReSubscriptions) {
+                    if (StringUtils.isNotEmpty(api.getPreviousId())) {
+                        List<Subscription> subscriptions = getApiSubscriptionDAO().getAPISubscriptionsByAPI(api
+                                .getPreviousId());
+                        for (Subscription subscription : subscriptions) {
+                            if (api.getPolicies().contains(subscription.getSubscriptionTier())) {
+                                if (!APIMgtConstants.SubscriptionStatus.ON_HOLD.equals(subscription.getStatus())) {
+                                    getApiSubscriptionDAO().addAPISubscription(UUID.randomUUID().toString(), api
+                                                    .getId(),
+                                            subscription.getApplication().getId(), subscription.getSubscriptionTier(),
+                                            subscription.getStatus());
+                                }
+                            } else {
+                                skippedSubscriptionList.add(subscription);
```

Comment:
We dont need to notify of skipped subscriptions due removal of tier here. Ideally we should notify user when the tier is being removed that they will loose subscriptions if they already exist